### ISSUE TITLE
NPE from SqlDateSerializer as _useTimestamp is not check for beeing null

### ISSUE THREAD

**Issue by mrowkow** [FasterXML/jackson-databind#774]
During serialization I've got the following error:
```
Caused by: java.lang.NullPointerException
        at com.fasterxml.jackson.databind.ser.std.SqlDateSerializer.serialize(SqlDateSerializer.java:48) [jackson-databind-2.5.2.jar:2.5.2]
        at com.fasterxml.jackson.databind.ser.std.SqlDateSerializer.serialize(SqlDateSerializer.java:18) [jackson-databind-2.5.2.jar:2.5.2]
```

It is thrown from the method: 
```
public void serialize(java.sql.Date value, JsonGenerator jgen, SerializerProvider provider)
        throws IOException, JsonGenerationException
    {
        if (_useTimestamp) {
            jgen.writeNumber(_timestamp(value));
        } else {
        	jgen.writeString(value.toString());
        }
    }
```
because the `_useTimestamp` is `null`

It set in DateTimeSerializerBase#createContextual by the framgment (asNumber is null):
```
        		Boolean asNumber = (format.getShape() == JsonFormat.Shape.STRING) ? Boolean.FALSE : null;
                // If not, do we have a pattern?
                TimeZone tz = format.getTimeZone();
                if (format.hasPattern()) {
                    String pattern = format.getPattern();
                    final Locale loc = format.hasLocale() ? format.getLocale() : prov.getLocale();
                    SimpleDateFormat df = new SimpleDateFormat(pattern, loc);
                    if (tz == null) {
                        tz = prov.getTimeZone();
                    }
                    df.setTimeZone(tz);
                    return withFormat(asNumber, df);
                }

```

**Comment by cowtowncoder** [comment]
Sounds like a bug indeed, thank you for reporting this.

**Comment by cowtowncoder** [comment]
Fixed for 2.5.4 (no more 2.4 releases planned)
